spec/components/candidate_interface/pending_gcse_required_component_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::PendingGcseRequiredComponent, type: :component do
+  let(:application_form) { create(:application_form) }
+
+  let(:course_option1) { create(:course_option, course: create(:course, :open_on_apply, accept_pending_gcse: true)) }
+  let(:course_option2) { create(:course_option, course: create(:course, :open_on_apply, accept_pending_gcse: false)) }
+
+  let(:application_choice1) do
+    build_stubbed(
+      :application_choice,
+      status: :unsubmitted,
+      course_option: course_option1,
+      application_form: application_form,
```

Comment:
sorry one more not. I don't think we need to pass in an application form here as the component renders based off of the passed in application choice 


```
result = render_inline(described_class.new(application_choice1))
```
so the default app form created by the factory should be fine

Reply:
I think this is because the `find_pending_uk_gcses(application_choice)` method on the component finds all relevant associated qualls with the application_form. 

I'm not sure it will work without stubbing the application form here as it's the point of connection?

Reply:
Right yea, good point. You can stub it out if you want, but it's probs not worth the hassle since you can use build_stubbed 👍